modules/openapi-generator/src/main/resources/cpp-rest-sdk-client/modelbase-header.mustache

```diff
@@ -162,10 +162,7 @@ template <typename T>
 utility::string_t ModelBase::toString( const std::shared_ptr<T>& val )
 {
     utility::stringstream_t ss;
-    if( val != nullptr )
-    {
-        val->toJson().serialize(ss);
-    }
```

Comment:
@SaverioCode thanks for the PR

quick question if you don't mind. why remove the null check here and below (line 208)?

Reply:
> @SaverioCode thanks for the PR
> 
> quick question if you don't mind. why remove the null check here and below (line 208)?

Hi!

I removed them in several places because they are reference and in c++ they can't be `nullptr` for definition. There is a compile error if a reference is `nullptr`. [A reference is an alias to an already-existing object or function](https://en.cppreference.com/w/cpp/language/reference). So in this scenario the `shared_ptr` can't be `nullptr`, what could actually be invalid is the pointer inside the `shared_ptr`, the content of the referred object, but not the object itself
As it's done for any object that contains pointers, I suppose it would make sense to check the content of the `shared_ptr` (it's not what was happening), but being a `shared_ptr` I would assume that the pointer inside of it's valid or I would think more of a program logic error. But this last statement can be different from team to team

Reply:
thanks for the explanation

cc @etherealjoy who added the nullptr check to see if he's further comment/question on this